builder/steps/gen-dockerfile/contents/test/generate_files_test.ts

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as util from 'util';
+
+const shellEscape: (args: Array<string>) => string = require('shell-escape');
```

Comment:
Prefer the simple "string[]" over Array<string> (You have few of them in different places).

We prefer using Array<> only when there is a union type like Array<string|null>.